src/php/tests/MemoryLeakTest/MemoryLeakTest.php

```diff
@@ -0,0 +1,2310 @@
+    'localhost:0',
+    [
+        'force_new' => true,
+        'credentials' => Grpc\ChannelCredentials::createInsecure()
+    ]
+);
+$interceptor_channel = Grpc\Interceptor::intercept($channel, new Grpc\Interceptor());
+$now = Grpc\Timeval::now();
+$deadline = $now->add(new Grpc\Timeval(100*1000));
+$state = $interceptor_channel->watchConnectivityState(1, $deadline);
+assert($state);
+unset($time);
+unset($deadline);
+$channel->close();
+
+// Test InterceptorChannelClose
+$channel = new Grpc\Channel(
+    'localhost:0',
+    [
+        'force_new' => true,
+        'credentials' => Grpc\ChannelCredentials::createInsecure()
+    ]
+);
+$interceptor_channel = Grpc\Interceptor::intercept($channel, new Grpc\Interceptor());
+assert($interceptor_channel != NULL);
+$channel->close();
+
+// Test InterceptorChannelGetTarget
+$channel = new Grpc\Channel(
+    'localhost:8888',
+    [
+        'force_new' => true,
+        'credentials' => Grpc\ChannelCredentials::createInsecure()
+    ]
+);
+$interceptor_channel = Grpc\Interceptor::intercept($channel, new Grpc\Interceptor());
+$target = $interceptor_channel->getTarget();
+assert(is_string($target));
+
+$channel->close();
+unset($server);
+
+
+//============== CallInvoker Test ====================
+class CallInvokerSimpleRequest
+{
+    private $data;
+    public function __construct($data)
+    {
+        $this->data = $data;
+    }
+    public function setData($data)
+    {
+        $this->data = $data;
+    }
+    public function serializeToString()
+    {
+        return $this->data;
+    }
+}
+
+class CallInvokerClient extends Grpc\BaseStub
+{
+
+  /**
+   * @param string $hostname hostname
+   * @param array $opts channel options
+   * @param Channel|InterceptorChannel $channel (optional) re-use channel object
+   */
+  public function __construct($hostname, $opts, $channel = null)
+  {
+    parent::__construct($hostname, $opts, $channel);
+  }
+
+  /**
+   * A simple RPC.
+   * @param SimpleRequest $argument input argument
+   * @param array $metadata metadata
+   * @param array $options call options
+   */
+  public function UnaryCall(
+    CallInvokerSimpleRequest $argument,
+    $metadata = [],
+    $options = []
+  ) {
+    return $this->_simpleRequest(
+      '/dummy_method',
+      $argument,
+      [],
+      $metadata,
+      $options
+    );
+  }
+}
+
+class CallInvokerUpdateChannel implements \Grpc\CallInvoker
+{
+    private $channel;
+
+    public function getChannel() {
+        return $this->channel;
+    }
+
+    public function createChannelFactory($hostname, $opts) {
+        $this->channel = new \Grpc\Channel('localhost:50050', $opts);
+        return $this->channel;
+    }
+
+    public function UnaryCall($channel, $method, $deserialize, $options) {
+        return new UnaryCall($channel, $method, $deserialize, $options);
+    }
+
+    public function ClientStreamingCall($channel, $method, $deserialize, $options) {
+        return new ClientStreamingC
```

Comment:
Why are there apparent version control markers in this file? This is blocking the import. We should probably have an OSS sanity check for this too, but this doesn't look right.

Reply:
sorry, this is my bad.